### ISSUE TITLE
 Method 'get_LockReleaseBehavior' in type 'Pomelo.EntityFrameworkCore.MySql.Migrations.Internal.MySqlHistoryRepository' from assembly 'Pomelo.EntityFrameworkCore.MySql, Version=9.0.0.0, Culture=neutral, PublicKeyToken=2cc498582444921b' does not have an implementation.

### ISSUE THREAD

**Issue by habtard** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1954]
### Steps to reproduce
```    
<ItemGroup>
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="9.0.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Abstractions" Version="9.0.0" />
    <PackageReference Include="Pomelo.EntityFrameworkCore.MySql" Version="9.0.0-preview.1" />
    <PackageReference Include="MySqlConnector" Version="2.3.5" />
</ItemGroup>
```

### The issue
Describe what is not working as expected.

If you are seeing an exception, include the full exceptions details (message and stack trace).

```
System.TypeLoadException: Method 'get_LockReleaseBehavior' in type 'Pomelo.EntityFrameworkCore.MySql.Migrations.Internal.MySqlHistoryRepository' from assembly 'Pomelo.EntityFrameworkCore.MySql, Version=9.0.0.0, Culture=neutral, PublicKeyToken=2cc498582444921b' does not have an implementation.
   at Microsoft.Extensions.DependencyInjection.MySqlServiceCollectionExtensions.AddEntityFrameworkMySql(IServiceCollection serviceCollection)
   at Pomelo.EntityFrameworkCore.MySql.Infrastructure.Internal.MySqlOptionsExtension.ApplyServices(IServiceCollection services)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.ApplyServices(IDbContextOptions options, ServiceCollection services)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.<GetOrAdd>g__BuildServiceProvider|4_1(IDbContextOptions _, ValueTuple`2 arguments)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.<>c.<GetOrAdd>b__4_0(IDbContextOptions contextOptions, ValueTuple`2 tuples)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd[TArg](TKey key, Func`3 valueFactory, TArg factoryArgument)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.GetOrAdd(IDbContextOptions options, Boolean providerRequired)
   at Microsoft.EntityFrameworkCore.DbContext..ctor(DbContextOptions options)
   at Sadie.Database.SadieContext..ctor(DbContextOptions`1 options) in /home/ash/Documents/SadieEmulator/Sadie.Database/SadieContext.cs:line 30
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
   at System.Reflection.MethodBaseInvoker.InvokeWithOneArg(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitDisposeCache(ServiceCallSite transientCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.DynamicServiceProviderEngine.<>c__DisplayClass2_0.<RealizeService>b__0(ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(ServiceIdentifier serviceIdentifier, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)

```

### Further technical details

MySQL version: 
Operating system: 
Pomelo.EntityFrameworkCore.MySql version:
Microsoft.AspNetCore.App version:

Other details about my project setup:

**Comment by MiklosPathy** [comment]
Have the same problem. It is a runtime error, thrown at DBContext constructor, when i try to start the application and it receives the connection string from the options object (appsettings.json)
Simply tried to update the EF core and they Pomelo driver from 8 to 9.

**Comment by CliffConway** [comment]
I also have this problem after upgrading to Net 9.0 and Pomelo preview.1. 

However, it is also happening with MySql 8.0.34 version

Does anyone know of a workaround or when this problem of a missing implementation is likely to take to resolve by Pomelo (I know - how long is a piece of string ;-)  )

**Comment by jolopher** [comment]
I have solved this problem by referencing the Microsoft.EntityFrameworkCore packages with version "9.0.0-preview.1.24081.2"
If I use the version 9.0.0 I get the same error.

**Comment by CliffConway** [comment]
Thank you for you solution. 
 
Are you referring to Microsoft.EntityFrameworkCore "9.0.0-preview.1.24081.2" or a nightly build of Pomelo.EntityframeworkCore "9.0.0-preview.1.24081.2". When I use the Microsoft one i get installation errors:-
 
NU1608: Detected package version outside of dependency constraint: Pomelo.EntityFrameworkCore.MySql 9.0.0-preview.1 requires Microsoft.EntityFrameworkCore.Relational (>= 9.0.0-preview.1.24081.2 && <= 9.0.0-preview.1.999999) but version Microsoft.EntityFrameworkCore.Relational 9.0.0 was resolved.
NU1608: Detected package version outside of dependency constraint: Npgsql.EntityFrameworkCore.PostgreSQL 7.0.0 requires Microsoft.EntityFrameworkCore (>= 7.0.0 && < 8.0.0) but version Microsoft.EntityFrameworkCore 9.0.0-preview.1.24081.2 was resolved.
NU1608: Detected package version outside of dependency constraint: Npgsql.EntityFrameworkCore.PostgreSQL 7.0.0 requires Microsoft.EntityFrameworkCore.Abstractions (>= 7.0.0 && < 8.0.0) but version Microsoft.EntityFrameworkCore.Abstractions 8.0.10 was resolved.
NU1608: Detected package version outside of dependency constraint: Npgsql.EntityFrameworkCore.PostgreSQL 7.0.0 requires Microsoft.EntityFrameworkCore.Relational (>= 7.0.0 && < 8.0.0) but version Microsoft.EntityFrameworkCore.Relational 9.0.0 was resolved.
NU1701: Package 'PresentationFramework 4.6.0' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework 'net9.0'. This package may not be fully compatible with your project.
NuGet\Install-Package : NU1605: Warning As Error: Detected package downgrade: Microsoft.EntityFrameworkCore.Abstractions from 9.0.0-preview.1.24081.2 to 
8.0.10. Reference the package directly from the project to select a different version. 
 EDWDataLibrary -> Microsoft.EntityFrameworkCore 9.0.0-preview.1.24081.2 -> Microsoft.EntityFrameworkCore.Abstractions (>= 9.0.0-preview.1.24081.2) 
 EDWDataLibrary -> Microsoft.EntityFrameworkCore.Abstractions (>= 8.0.10)
At line:1 char:1
+ NuGet\Install-Package Microsoft.EntityFrameworkCore -Version 9.0.0-pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.InstallPackageCommand
 
NuGet\Install-Package : NU1605: Warning As Error: Detected package downgrade: Microsoft.EntityFrameworkCore from 9.0.0 to 9.0.0-preview.1.24081.2. Reference 
the package directly from the project to select a different version. 
 EDWDataLibrary -> Microsoft.EntityFrameworkCore.Relational 9.0.0 -> Microsoft.EntityFrameworkCore (>= 9.0.0) 
 EDWDataLibrary -> Microsoft.EntityFrameworkCore (>= 9.0.0-preview.1.24081.2)
At line:1 char:1
+ NuGet\Install-Package Microsoft.EntityFrameworkCore -Version 9.0.0-pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.InstallPackageCommand
 
NuGet\Install-Package : Package restore failed. Rolling back package changes for 'EDWDataLibrary'.
At line:1 char:1
+ NuGet\Install-Package Microsoft.EntityFrameworkCore -Version 9.0.0-pr ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Install-Package], Exception
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.InstallPackageCommand

**Comment by jhouzvicka** [comment]
This is a duplicate of https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/1944.

**Comment by og1sha** [comment]
> I have solved this problem by referencing the Microsoft.EntityFrameworkCore packages with version "9.0.0-preview.1.24081.2" If I use the version 9.0.0 I get the same error.

For me the error persists even when downgrading packages.

**Comment by jolopher** [comment]
@CliffConway this is what worked for me (I kept only relevant package references)

```
<PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="9.0.0-preview.1.24081.2">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
</PackageReference>
<PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="9.0.0-preview.1.24081.2">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
</PackageReference>
<PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="9.0.0-preview.1.24081.2" />
<PackageReference Include="MySqlConnector" Version="2.4.0" />
<PackageReference Include="Pomelo.EntityFrameworkCore.MySql" Version="9.0.0-preview.1" />
```

After all, I will wait until the final release to move to .NET 9 to avoid future issues.

**Comment by lauxjpn** [comment]
You are trying to use our `9.0.0-preview.1` release (that is only compatible with EF Core `9.0.0-preview.1`) with EF Core `9.0.x`.
We have a EF Core `9.0.x` compatible prerelease in our [nightly build feeds](https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql?tab=readme-ov-file#nightly-builds). See https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/issues/1841#issuecomment-2525312763 and subsequent posts.

**Comment by ikumshyamal** [comment]
I have resolved this by installing the Pomelo.EntityFrameworkCore.MySql --version 9.0.0-preview.2

dotnet add package Pomelo.EntityFrameworkCore.MySql --version 9.0.0-preview.2.efcore.9.0.0

**Comment by waltermillan** [comment]
I solving this using:

dotnet add package Microsoft.EntityFrameworkCore --version 9.0.2
dotnet add package Pomelo.EntityFrameworkCore.MySql --version 9.0.0-preview.2.efcore.9.0.0

Thanks to Ikum Shyamal!!

**Comment by lauxjpn** [comment]
@waltermillan We released `9.0.0-preview.3.efcore.9.0.0` in the meantime, so you likely want to reference that version instead.
